src/transformStateWithClones.js

```diff
@@ -1,13 +1,31 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const current = { ...state };
+  const resultingArray = [];
+  let newr;
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const prop in actions[i].extraData) {
+        current[prop] = actions[i].extraData[prop];
+      }
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const word in actions[i].keysToRemove) {
+        delete current[actions[i].keysToRemove[word]];
+      }
+    }
+
+    if (actions[i].type === 'clear') {
+      for (const key in current) {
+        delete current[key];
+      }
+    } newr = { ...current }; resultingArray[i] = newr;
```

Comment:
Never do like this!!!
You put 2 command in the line with a closing `}`
Why?!!!!

Reply:
Why do you need `newr`?

Reply:
When I was trying to use only const current, my solution gave three three identical lines, which were the result of all the performed manipulations with the input . Second object has fixed this, so I have got right solving for all tests. 
I`ll try do to that in another way